### PR TITLE
feat: annotations on gateway serviceaccount

### PR DESCRIPTION
**Pull request by bryankaraffa**
We are attempting to implement [IAM Roles for Service Accounts on AWS EKS [IRSA]](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to our clusters which use Istio Ingress Gateways.  In order to associate the ServiceAccount resource with the AWS IAM Role, the ServiceAccount [needs to have a certain annotation](https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html).

This PR adds values to the istio gateway helm charts [ingress and egress] to allow annotations to be added to the service account, which can be used to enable IRSA for the Istio Gateway resources in the cluster.  I implemented this based on the implementation from the `helm chart create` boilerplate.

This is the minimum changes required to allow using the official helm charts and IRSA.  May be useful for other use cases too where istio gateway service account needs to be annotated.

As additional PRs, we can also provide enhancements to the charts which would allow toggling ServiceAccount creation, and explicitly defining the ServiceAccount name.  This enables use-cases where the ServiceAccount that the user wants to use for Istio Gateway is created/managed separately from the helm release.  It would also bring these charts in-line with what comes from the starter scaffold created by [`helm create <chart name>`](https://helm.sh/docs/helm/helm_create/)

put an X in all the areas that this PR affects.

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by bryankaraffa** [comment]
@nrjpoddar -- Looking at the Environment WG Roadmap, seems like you may have an interest in this PR

File: manifests/charts/gateways/istio-egress/values.yaml [hunk 1]
```diff
@@ -59,6 +59,11 @@ gateways:
 
     configVolumes: []
     additionalContainers: []
+
+    serviceAccount:
+      # Annotations to add to the service account
+      annotations: {}
```
Comment (nrjpoddar) [inline manifests/charts/gateways/istio-egress/values.yaml:65]: This is an array? If yes, should we initialize it to `[]`?

**Review by nrjpoddar — COMMENTED** [review]
Minor comment but rest looks good to me.

File: manifests/charts/gateways/istio-egress/values.yaml (hunk 1 above)
Comment (bryankaraffa) [reply manifests/charts/gateways/istio-egress/values.yaml:65]: Annotations, like labels, are key/value maps

https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#attaching-metadata-to-objects

Comment (nrjpoddar) [reply manifests/charts/gateways/istio-egress/values.yaml:65]: Yes, of course. Not sure why I had list on my mind when reviewed it first time around.

**Review by nrjpoddar — APPROVED** [review]

**Comment by bryankaraffa** [comment]
@costinm @ostromart @sdake -- If approved, I'm hoping to get an `ok-to-test` label and move this forward please.  Thanks in advance!

**Comment by bryankaraffa** [comment]
Unfortunately I'm not too familiar with go lang, and fixing the structure / unit test has proven a bit difficult

https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/33914/unit-tests_istio/1422998885625761792#1:build-log.txt%3A8185
`validate_values_test.go:229: file /home/prow/go/src/istio.io/istio/manifests/charts/gateways/istio-egress/values.yaml failed validation with: unknown field "serviceAccount" in v1alpha1.EgressGatewayConfig`

I attempted to add `serviceAccount` ti the values_types.proto in 8df6cd9 but that seems to have no affect.  Any suggestions or help from community to get this across the finish line would be very much appreciated!

**Comment by bryankaraffa** [comment]
Ready for final reviews, approvals, and hopefully a merge! 🚀

**Comment by bryankaraffa** [comment]
@costinm @jacob-delgado request for review as a code owner.  Thanks in advance!

**Review by jacob-delgado — APPROVED** [review]

**Comment by bryankaraffa** [comment]
@googlebot I consent

**Comment by bryankaraffa** [comment]
:shipit:

**Review by Xunzhuo — APPROVED** [review]
LGTM! Thanks a lot!

**Review by shamsher31 — APPROVED** [review]

**Comment by jacob-delgado** [comment]
cc @costinm @Monkeyanator I went through a double checked this against `helm create` and this is what is done there. Looking for more approvals.

**Review by howardjohn — APPROVED** [review]

**Review by Monkeyanator — APPROVED** [review]

**Pull request merged**